File: Python/Customs/CustomPOSTags.py
#-*- coding: utf-8 -*-
#!/usr/bin/env python

# This example shows how to use LTP API to perform POS taggings on the custom
# word segmentation result.

from LTML import LTML
import sys
import urllib, urllib2

def ParsingWithCustomPostags():
    ltml = LTML()
    ltml.build_from_words([("这",       "r"),
                           ("是",       "v"),
                           ("自定义",   "a"),
                           ("分词",     "n"),
                           ("结果",     "n"),
                           ("的",       "u"),
                           ("示例",     "n")])
    xml  = ltml.tostring()

    uri_base = "http://api.ltp-cloud.com/analysis/?"

    data = {
            "api_key"   : "YourApiKey",
            "text"      : xml,
            "format"    : "conll",
            "pattern"   : "dp",
            "xml_input" : "true"
            }

    params = urllib.urlencode(data)

    try:
        request  = urllib2.Request(uri_base)
        response = urllib2.urlopen(request, params)
        content  = response.read().strip()
        print content
    except urllib2.HTTPError, e:
        print >> sys.stderr, e.reason

if __name__=="__main__":
    ParsingWithCustomPostags()

File: Ruby/Customs/CustomPOSTags.rb
# coding: UTF-8
require 'net/http'
require 'uri'
require './LTML'

ltml = LTML.new
ltml.build_from_words_with_postags([['我', 'r'],
                                    ['爱', 'v'],
                                    ['北京', 'ns'],
                                    ['天安门', 'ns']], 'UTF-8')

params = {
  :api_key    => 'YourApiKey',
  :text       => ltml.to_s,
  :format     => 'conll',
  :pattern    => 'dp',
  :xml_input  => 'true'
}

url = URI.parse("http://api.ltp-cloud.com/analysis/")

response = Net::HTTP.post_form(url, params)

puts response.body
